From e06f0b67b6530db6788e85f98f7200ac6de3dc2b Mon Sep 17 00:00:00 2001
From: Paul Szabo <psz@maths.usyd.edu.au>
Date: Fri, 14 Aug 2015 14:59:56 -0400
Subject: [PATCH] session-record: write ut_line for remote logins too

We currently only write out ut_line for local logins.

This commit makes sure it gets written out for remote
logins as well.

https://bugzilla.gnome.org/show_bug.cgi?id=599103
---
 daemon/gdm-session-record.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/daemon/gdm-session-record.c b/daemon/gdm-session-record.c
index 4ed60a1..bbd9c54 100644
--- a/daemon/gdm-session-record.c
+++ b/daemon/gdm-session-record.c
@@ -158,62 +158,61 @@ record_set_host (UTMP       *u,
                 hostname = g_strdup (x11_display_name);
         }
 
         if (hostname != NULL) {
                 strncpy (u->ut_host, hostname, sizeof (u->ut_host));
                 g_debug ("using ut_host %.*s", (int) sizeof (u->ut_host), u->ut_host);
                 g_free (hostname);
 
 #ifdef HAVE_UT_UT_SYSLEN
                 u->ut_syslen = MIN (strlen (hostname), sizeof (u->ut_host));
 #endif
         }
 #endif
 }
 
 static void
 record_set_line (UTMP       *u,
                  const char *display_device,
                  const char *x11_display_name)
 {
         /*
          * Set ut_line to the device name associated with this display
          * but remove the "/dev/" prefix.  If no device, then use the
          * $DISPLAY value.
          */
         if (display_device != NULL
             && g_str_has_prefix (display_device, "/dev/")) {
                 strncpy (u->ut_line,
                          display_device + strlen ("/dev/"),
                          sizeof (u->ut_line));
-        } else if (x11_display_name != NULL
-                   && g_str_has_prefix (x11_display_name, ":")) {
+        } else if (x11_display_name != NULL) {
                 strncpy (u->ut_line,
                          x11_display_name,
                          sizeof (u->ut_line));
         }
 
         g_debug ("using ut_line %.*s", (int) sizeof (u->ut_line), u->ut_line);
 }
 
 void
 gdm_session_record_login (GPid                  session_pid,
                           const char           *user_name,
                           const char           *host_name,
                           const char           *x11_display_name,
                           const char           *display_device)
 {
         UTMP        session_record = { 0 };
         UTMP       *u;
 
         record_set_username (&session_record, user_name);
 
         g_debug ("Writing login record");
 
 #if defined(HAVE_UT_UT_TYPE)
         session_record.ut_type = USER_PROCESS;
         g_debug ("using ut_type USER_PROCESS");
 #endif
 
         record_set_timestamp (&session_record);
         record_set_pid (&session_record, session_pid);
 
-- 
2.7.4

